DataHub.js

```diff
@@ -0,0 +1,299 @@
+/*!
+ * Copyright (c) 2018-2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import axios from 'axios';
+import {Cipher} from './Cipher.js';
+import {IndexHelper} from './IndexHelper.js';
+
+export class DataHub {
+  /**
+   * Creates a new DataHub instance. The storage for the data hub must already
+   * exist and have an HTTPS API at the given `baseUrl`.
+   *
+   * @param {String} baseUrl an optional endpoint for the data hub storage.
+   * @param {Object} config the data hub's configuration document.
+   * @param {kek} kek a KEK API for wrapping content encryption keys.
+   * @param {hmac} hmac an HMAC API for blinding indexable attributes.
+   *
+   * @return {DataHub}.
+   */
+  constructor({baseUrl = '/data-hubs', config, kek, hmac}) {
+    this.config = config;
+    this.kek = kek;
+    // TODO: support passing cipher `version`
+    this.cipher = new Cipher();
+    this.indexHelper = new IndexHelper({hmac});
+    const root = `${baseUrl}/${encodeURIComponent(config.id)}`;
+    this.urls = {
+      root,
+      documents: `${root}/documents`,
+      query: `${root}/query`
+    };
+  }
+
+  /**
+   * Ensures that future documents inserted or updated using this DataHub
+   * instance will be indexed according to the given attribute, provided that
+   * they contain that attribute.
+   *
+   * @param {Array|Object} attribute the attribute name or an array of
+   *   attribute names.
+   * @param {Boolean} unique `true` if attribute values should be considered
```

Comment:
```suggestion
   * @param {Boolean} [unique=false] `true` if attribute values should be considered
```
And update verbiage.